src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -2,143 +2,183 @@
+            {
+                // If guid is not identified by vs-solutionpersistence, check in project element itself
+                var guid = projectRootElement.GetProjectTypeGuid();
+                if (string.IsNullOrEmpty(guid))
+                {
+                    Reporter.Error.WriteLine(CommonLocalizableStrings.UnsupportedProjectType, Path.GetFullPath(relativePath));
+                    return;
+                }
+                project = solution.AddProject(relativePath, guid, solutionFolder);
+            }
+            // Generate intermediate solution folders
+            if (solutionFolder is null && !_inRoot)
+            {
+                var relativePathDirectory = Path.GetDirectoryName(relativePath);
+                if (!string.IsNullOrEmpty(relativePathDirectory))
+                {
+                    SolutionFolderModel relativeSolutionFolder = solution.AddFolder(GetSolutionFolderPathWithForwardSlashes(relativePathDirectory));
+                    project.MoveToFolder(relativeSolutionFolder);
+                    // Avoid duplicate folder/project names
+                    if (project.Parent is not null && project.Parent.ActualDisplayName == project.ActualDisplayName)
+                    {
+                        solution.RemoveFolder(project.Parent);
+                    }
+                }                
+            }
+            // Add settings based on existing project instance
+            ProjectInstance projectInstance = new ProjectInstance(projectRootElement);
+            string projectInstanceId = projectInstance.GetProjectId();
+            if (!string.IsNullOrEmpty(projectInstanceId))
+            {
+                project.Id = new Guid(projectInstanceId);
+            }
+            var projectInstanceBuildTypes = projectInstance.GetConfigurations();
+            var projectInstancePlatforms = projectInstance.GetPlatforms();
 
-        private static string TrimProject(string path)
-        {
-            return Path.GetDirectoryName(path);
-        }
+            foreach (var solutionPlatform in solution.Platforms)
+            {
+                var projectPlatform = projectInstancePlatforms.FirstOrDefault(
+                    x => x.Replace(" ", string.Empty) ==  solutionPlatform.Replace(" ", string.Empty), projectInstancePlatforms.FirstOrDefault("Any CPU"));
+                project.AddProjectConfigurationRule(new ConfigurationRule(BuildDimension.Platform, "*", solutionPlatform, projectPlatform));
+            }
 
-        private static string TrimProjectDirectory(string path)
-        {
-            return Path.GetDirectoryName(path);
+            foreach (var solutionBuildType in solution.BuildTypes)
+            {
+                var projectBuildType = projectInstanceBuildTypes.FirstOrDefault(
+                    x => x.Replace(" ", string.Empty) == solutionBuildType.Replace(" ", string.Empty), projectInstanceBuildTypes.FirstOrDefault("Debug"));
```

Comment:
This exists as the current behavior, however this is more of a different implementation on our end.